### DISCUSSION TITLE
Is ws.send thread safe?I want to call it in sub thread

### DISCUSSION THREAD

**Discussion by seetimee** [miguelgrinberg/flask-sock#86] [Q&A] [answered]

**Comment by miguelgrinberg** [comment, marked as answer]
Yes, you can pass the `ws` object to another thread and use it there. But if you plan to use the same `ws` object from multiple threads that are running at the same time then the answer is more complicated, I'd say that is probably best avoided or protected with a lock to ensure concurrent writes cannot interfere with each other.

**Reply by seetimee** [reply]
Can I derectly rewrite simple_websocket.ws.Base.send by give the function a Instance lock.

**Reply by seetimee** [reply]
I indeed use it in multi thread scenario

**Reply by miguelgrinberg** [reply]
I do not want to complicate this package with locking, as most people do not need it and it can have performance implications. But if you use your own fork you are obviously free to modify the send function however you want.

**Reply by seetimee** [reply]
thans a lot for your reply🤗
